client/dashboard/sites/settings/index.tsx

```diff
@@ -5,17 +5,17 @@ import {
 	Card,
 } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { siteSettingsQuery } from '../../app/queries';
+import { siteFeaturesQuery, siteSettingsQuery } from '../../app/queries';
 import { siteRoute } from '../../app/router';
 import PageLayout from '../../components/page-layout';
-import RouterLinkSummaryButton from '../../components/router-link-summary-button';
-import { getSubscriptionGiftingSettingBadges } from '../settings-subscription-gifting';
+import SubscriptionGiftingSettingsSummary from '../settings-subscription-gifting/summary';
 
 export default function SiteSettings() {
 	const { siteSlug } = siteRoute.useParams();
+	const { data: features } = useQuery( siteFeaturesQuery( siteSlug ) );
```

Comment:
While I have a question above about the need to call this endpoint or not, I don't see this query added to the "loader" which mean it won't break the rendering of the page causing bad UX. (rendering nothing until it loads). This is a pattern that we should be using for any mandatory REST API call (preloading in the loader).

Reply:
I don't have much experience in TanStack stack. However, it seems we already returning nothing when the data has not fully loaded, e.g.:

https://github.com/Automattic/wp-calypso/blob/559471515200918b15d2a4f38bf5768ced716d76/client/dashboard/me/profile/index.tsx#L127-L129

https://github.com/Automattic/wp-calypso/blob/559471515200918b15d2a4f38bf5768ced716d76/client/dashboard/sites/settings-subscription-gifting/index.tsx#L47-L49

So my question is:

- do we still need to add the queries to the loader, and
- if yes, do we still need the above conditionals in the components?

Reply:
I think I see the difference... if I don't put the queries in the loader, I am getting multiple `W` logo flashes... cool OK I'll update this!

Reply:
Yes, we still need to add it to the router loader, because Tanstack Router will wait for page navigation until all the basic data needed for the route has been fetched.

Reply:
> if yes, do we still need the above conditionals in the components?

It's not needed, but TS requires it.

Reply:
Thanks, done: https://github.com/Automattic/wp-calypso/pull/103227/commits/be0f083ca57ec45793b9fc1daa2a72eeddca468e